Objects/setobject.c

```diff
@@ -1515,17 +1529,21 @@ set_difference(PySetObject *so, PyObject *other)
         while (set_next(so, &pos, &entry)) {
             key = entry->key;
             hash = entry->hash;
+            Py_INCREF(key);
             rv = _PyDict_Contains(other, key, hash);
```

Comment:
Note: cherry-picker only struggled because this is `_PyDict_Contains_KnownHash()` in `main`.